src/gridstack-engine.ts

```diff
@@ -160,28 +160,30 @@ export class GridStackEngine {
 
     let collide: GridStackNode;
     let overMax = 0.5; // need >50%
-    collides.forEach(n => {
-      if (n.locked || !n._rect) return;
+    for (let n of collides) {
+      if (n.locked || !n._rect) {
+        break;
```

Comment:
of course, such a simple fix. if any are locked bailed out of the entire loop...